lib/services/plugins/npm-plugins-service-base.ts

```diff
@@ -10,33 +10,43 @@ import temp = require("temp");
 temp.track();
 
 export abstract class NpmPluginsServiceBase implements IPluginsService {
+	private static NPM_REGISTRY_ADDRESS = "http://registry.npmjs.org";
+
 	constructor(protected $errors: IErrors,
 		protected $logger: ILogger,
 		protected $prompter: IPrompter,
 		protected $fs: IFileSystem,
 		protected $project: Project.IProject,
 		protected $projectConstants: Project.IConstants,
 		protected $childProcess: IChildProcess,
+		protected $httpClient: Server.IHttpClient,
 		private $hostInfo: IHostInfo,
 		private $progressIndicator: IProgressIndicator) { }
 
 	public findPlugins(keywords: string[]): IFuture<IBasicPluginInformation[]> {
 		return ((): IBasicPluginInformation[] => {
 			let pluginsFound: IBasicPluginInformation[] = [];
 
-			let searchParams = ["search"].concat(keywords);
+			let query = this.composeSearchQuery(keywords);
+
+			let searchParams = ["search"].concat(query);
 
 			let npmCommand = this.$hostInfo.isWindows ? "npm.cmd" : "npm";
 
 			let npmFuture = this.$childProcess.spawnFromEvent(npmCommand, searchParams, "close");
 
 			this.$logger.printInfoMessageOnSameLine("Searching for plugins in npm, please wait.");
+
 			this.$progressIndicator.showProgressIndicator(npmFuture, 2000).wait();
 
 			let npmSearchResult = npmFuture.get();
 
 			if (npmSearchResult.stderr) {
-				this.$errors.failWithoutHelp(npmSearchResult.stderr);
+				// npm will write "npm WARN Building the local index for the first time, please be patient" to the stderr and if it is the only message on the stderr we should ignore it.
+				let splitError = npmSearchResult.stderr.split("\n");
+				if (splitError.length > 2 || !(splitError[0].indexOf("Building the local index for the first time") >= 0)) {
```

Comment:
instead of `!(splitError[0]...` you can use `|| splitError[0].indexOf("Building the local index for the first time") === -1)`